pkg/controller/statefulset/stateful_set_status_updater_test.go

```diff
@@ -144,3 +145,22 @@ func TestStatefulSetStatusUpdaterGetAvailableReplicas(t *testing.T) {
 		t.Errorf("UpdateStatefulSetStatus mutated the sets replicas %d", set.Status.AvailableReplicas)
 	}
 }
+
+func TestStatefulSetUpdaterUpdateExceedDeadline(t *testing.T) {
+	set := newStatefulSet(3)
+	status := apps.StatefulSetStatus{ObservedGeneration: 1, Replicas: 2}
+	fakeClient := &fake.Clientset{}
+	updater := NewRealStatefulSetStatusUpdater(fakeClient, nil)
+	fakeClient.AddReactor("update", "statefulsets", func(action core.Action) (bool, runtime.Object, error) {
+		update := action.(core.UpdateAction)
+		time.Sleep(15 * time.Second)
+		return true, update.GetObject(), nil
+	})
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	if err := updater.UpdateStatefulSetStatus(ctx, set, &status); err != nil {
+		if err != context.DeadlineExceeded {
+			t.Errorf("expected error: %#v, but got: %#v", context.DeadlineExceeded, err)
+		}
+	}
```

Comment:
```suggestion
	if err := updater.UpdateStatefulSetStatus(ctx, set, &status); err == nil ||  err != context.DeadlineExceeded {
			t.Errorf("expected error: %#v, but got: %#v", context.DeadlineExceeded, err)
	}
```

The test is working because is returning nil 

>     /home/aojea/go/src/k8s.io/kubernetes/pkg/controller/statefulset/stateful_set_status_updater_test.go:160: expected error: context.deadlineExceededError{}, but got: <nil>

Reply:
actually, the test can be simpler and not depend on timeout at all
```diff
diff --git a/pkg/controller/statefulset/stateful_set_status_updater_test.go b/pkg/controller/statefulset/stateful_set_status_updater_test.go
index 3e169d95957..6d0b7429809 100644
--- a/pkg/controller/statefulset/stateful_set_status_updater_test.go
+++ b/pkg/controller/statefulset/stateful_set_status_updater_test.go
@@ -20,7 +20,6 @@ import (
        "context"
        "errors"
        "testing"
-       "time"
 
        apps "k8s.io/api/apps/v1"
        apierrors "k8s.io/apimachinery/pkg/api/errors"
@@ -153,14 +152,11 @@ func TestStatefulSetUpdaterUpdateExceedDeadline(t *testing.T) {
        updater := NewRealStatefulSetStatusUpdater(fakeClient, nil)
        fakeClient.AddReactor("update", "statefulsets", func(action core.Action) (bool, runtime.Object, error) {
                update := action.(core.UpdateAction)
-               time.Sleep(15 * time.Second)
                return true, update.GetObject(), nil
        })
-       ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-       defer cancel()
-       if err := updater.UpdateStatefulSetStatus(ctx, set, &status); err != nil {
-               if err != context.DeadlineExceeded {
-                       t.Errorf("expected error: %#v, but got: %#v", context.DeadlineExceeded, err)
-               }
+       ctx, cancel := context.WithCancel(context.Background())
+       cancel()
+       if err := updater.UpdateStatefulSetStatus(ctx, set, &status); err == nil || err != context.DeadlineExceeded {
+               t.Errorf("expected error: %#v, but got: %#v", context.DeadlineExceeded, err)
        }
 }
```